TracNghiem/app/Providers/RepositoryServiceProvider.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Providers;
+
+use Illuminate\Support\ServiceProvider;
+
+class RepositoryServiceProvider extends ServiceProvider
+{
+
+	/**
+	 * Register any application services.
+	 *
+	 * @return void
+	 */
+	public function register()
+	{
+		//
+	}
+
+	/**
+	 * Bootstrap any application services.
+	 *
+	 * @return void
+	 */
+	public function boot()
+	{
+
+			app()->singleton(
+				"App\Repositories\Exam\ExamRepositoryInterface",
+				"App\Repositories\Exam\ExamRepository",
+				
+			);
+
+			app()->singleton(
+				"App\Repositories\Result\ResultRepositoryInterface",
+				"App\Repositories\Result\ResultRepository",
+			);
```

Comment:
nếu viết như này thì số lượng repo càng tăng thì cái file này sẽ ngày càng dài. 
chỗ này nên định nghĩa 1 array chứa tên các repo rồi dùng vòng for để chạy singleton. Sau này khi muốn thêm một repo mới thì chỉ cần thêm key name vào trong array là xong.